### PR TITLE
Use applyDefaultHierarchyTemplate

### LINKED ISSUE #12: apply kotlin.mpp.applyDefaultHierarchyTemplate=true
We might be using a legacy template.
I think we can migrate to a new one.

```
Consider removing dependsOn-calls or disabling the default template by adding
    'kotlin.mpp.applyDefaultHierarchyTemplate=false'
to your gradle.properties
```

### PR DESCRIPTION
**Pull request by kseito**
## Issue
- close #12 

## Overview (Required)
- Use applyDefaultHierarchyTemplate.
- Remove unnecessary definition.
- Following the above, the warning disappears when Gradle sync.

## Links
- https://kotlinlang.org/docs/multiplatform-hierarchy.html

## Screenshot (Optional if screenshot test is present or unrelated to UI)
Nothing.

## Movie (Optional)
Nothing.

### CODE REVIEW

File: build-logic/src/main/kotlin/io/github/droidkaigi/confsched/primitive/KmpIosPlugin.kt [hunk 1]
```diff
@@ -42,21 +43,7 @@ class KmpIosPlugin : Plugin<Project> {
                         }
                     }
                 }
-                with(sourceSets) {
```
Comment (kseito) [inline build-logic/src/main/kotlin/io/github/droidkaigi/confsched/primitive/KmpIosPlugin.kt:45]: supplement: 
According to [the official documentation](https://kotlinlang.org/docs/multiplatform-hierarchy.html#see-the-full-hierarchy-template), these are predefined and therefore unnecessary.

**Comment by takahirom** [comment]
Could you check if the iOS run?

* Run `make build-kmp-module` in the app-ios directory
* Open XCode by `make open`
* Run on iOS app simulator

**Comment by kseito** [comment]
>Could you check if the iOS app run with updated branch?

I followed the steps you wrote and verified that it works on my machine.

**Review by takahirom — APPROVED** [review]
Thank you for your check and contribution!

**Comment by takahirom** [comment]
@kseito Is there any reason you don't make this PR ready? 👀

**Comment by takahirom** [comment]
Thanks!

**Pull request merged**